EIPS/eip-modular-smart-contract-accounts-and-plugins.md

```diff
@@ -0,0 +1,352 @@
+---
+title: Modular Smart Contract Accounts and Plugins
+description: An interface for modular smart contract accounts with plugins.
+author: Adam Egyed (@adamegyed), Fangting Liu (@trinity-0111)
+discussions-to: https://ethereum-magicians.org/t/eip-modular-smart-contract-accounts-and-plugins/13885
+status: Draft
+type: Standards Track
+category: ERC
+created: 2023-04-18
+requires: 4337
+---
+
+## Abstract
+
+This proposal standardizes smart contract accounts and account plugins, which are smart contract interfaces that allow for composable logic within smart contract accounts. This proposal is compliant with [ERC-4337](./eip-4337.md), and builds on the existing work from [ERC-2535](./eip-2535.md) when defining interfaces for updating and querying modular function implementations.
+
+Goals:
+
+- Provide standards for how validation, execution, and hook functions for smart contract accounts should be written.
+- Provide standards for how compliant accounts should add, update, remove, and inspect plugins.
```

Comment:
Not _exactly_ suitable for the abstract. Perhaps this could go in the motivation?

Reply:
Moved to the end of motivation